<launch>
    <!-- LAUNCHES RVIZ, THE ROBOT STATE PUBLISHER AND THE JOINT STATE PUBLISHER -->

    <arg name="prefix" default=""/>
    <arg name="gui" default="true"/>
    <arg name="use_joint_publisher_gui" default="true"/>

    <group if="$(eval arg('gazebo_model_type')=='fixed_hip')">
        <param name="robot_description" command="$(find xacro)/xacro --inorder '$(find awesome_leg)/description/urdf/awesome_leg_fixed.xacro'" />
    </group>
    <group if="$(eval arg('gazebo_model_type')=='sliding_hip')">
        <param name="robot_description" command="$(find xacro)/xacro --inorder '$(find awesome_leg)/description/urdf/awesome_leg.xacro'" />
    </group>
    <group if="$(eval arg('gazebo_model_type')=='sliding_all')">
        <param name="robot_description" command="$(find xacro)/xacro --inorder '$(find awesome_leg)/description/urdf/awesome_leg_sliding_all.xacro'" />
    </group>
    
    <node if="$(arg gui)"
            type="rviz"
            name="rviz"
            pkg="rviz"
            args="-d $(find awesome_leg)/rviz/rviz_standalone.rviz"/>
            output="screen"/>
    
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" output="screen">
    <param name="publish_default_efforts" value="True"/>
    </node>

    <group if="$(arg use_joint_publisher_gui)">
        <node name="joint_state_publisher_gui" pkg="joint_state_publisher_gui" type="joint_state_publisher_gui" />  
    </group>

</launch>
